proxy/envoy/ingress.go

```diff
@@ -161,17 +159,19 @@ func generateIngress(conf *IngressConfig) *Config {
 		},
 	}
 
-	// configure for HTTPS if provided with a secret name
-	if conf.Secret != "" {
-		// configure Envoy
+	// configure for HTTPS if there is a secret
+	if secret, err := conf.Secrets.GetSecret("*"); err != nil {
```

Comment:
The name of the secret is hidden by the Secret interface. Instead of addressing secrets by secret name, they are addressed by host.